lib/rubocop/cop/style/float_division.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for division with integers coerced to floats.
+      # It is recommended to either always use `fdiv` or coerce left side only.
+      #
+      # @example EnforcedStyle: left_coerce (default)
+      #   # bad
+      #   a / b.to_f
+      #   a.to_f / b.to_f
+      #
+      #   # good
+      #   a.to_f / b
+      #
+      # @example EnforcedStyle: right_coerce
+      #   # bad
+      #   a.to_f / b
+      #   a.to_f / b.to_f
+      #
+      #   # good
+      #   a / b.to_f
+      #
+      # @example EnforcedStyle: fdiv
+      #   # bad
+      #   a / b.to_f
+      #   a.to_f / b
+      #   a.to_f / b.to_f
+      #
+      #   # good
+      #   a.fdiv(b)
+      class FloatDivision < Cop
+        include ConfigurableEnforcedStyle
+
+        def_node_matcher :right_coerce?, <<-PATTERN
+          (send _ :/ (send _ :to_f))
+        PATTERN
+        def_node_matcher :left_coerce?, <<-PATTERN
+          (send (send _ :to_f) :/ _)
+        PATTERN
+
+        def on_send(node)
+          add_offense(node) if offense_condition?(node)
+        end
+
+        private
+
+        def offense_condition?(node)
+          style == :left_coerce && right_coerce?(node) ||
+            style == :right_coerce && left_coerce?(node) ||
+            style == :fdiv && coerce?(node)
```

Comment:
I would prefer this method to look more similar to `#message`, to make it obvious that their logic is similar e.g.
```rb
def offense_condition?(node)
  case style
  when :left_coerce
    right_coerce?(node)
  when :right_coerce
    left_coerce?(node)
  when :fdiv
    right_coerce?(node) || left_coerce?(node)
  else
    false
  end
end
```

Reply:
I'm surprised we don't have a cop for that already. :-)

Reply:
Fixed.